features/emojiMod.js

```diff
@@ -0,0 +1,165 @@
+const staffRoles = ["mvp", "moderator", "admin", "admins"];
+
+const isStaff = member =>
+  (member.roles || []).some(role =>
+    staffRoles.includes(role.name.toLowerCase())
+  );
+
+const config = {
+  // This is how many ️️warning reactions a post must get until it's considered an official warning
+  warningThreshold: 1,
+  // This is how many ️️warning reactions a post must get until mods are alerted
+  thumbsDownThreshold: 2,
+  // This is how many ️️warning reactions a post must get the message is deleted
+  deletionThreshold: Infinity
+};
+
+const warningMessages = {};
+
+const thumbsDownEmojis = ["👎", "👎🏻", "👎🏼", "👎🏽", "👎🏾", "👎🏿"];
+
+const reactionHandlers = {
+  "⚠️": (bot, reaction, message, member) => {
+    if (!isStaff(member)) {
+      return;
+    }
+    const usersWhoReacted = reaction.users.map(user =>
+      message.guild.members.get(user.id)
+    );
+    const numberOfTotalReactions = usersWhoReacted.length;
+    const numberOfStaffReactions = usersWhoReacted.filter(isStaff).length;
+
+    const modLogChannel = bot.channels.find(
+      channel =>
+        channel.name === "mod-log" || channel.id === "257930126145224704"
+    );
+
+    const userNames = usersWhoReacted
+      .filter(user => !isStaff(user))
+      .map(member => member.user.username)
+      .join(", ");
+
+    const staffNames = usersWhoReacted
+      .filter(isStaff)
+      .map(member => member.user.username)
+      .join(", ");
+
+    let logMessage = "";
+    const logMessageEnding = [
+      "\n\n",
+      `\`${message.content}\``,
+      "\n\n",
+      `Link: https://discordapp.com/channels/${message.guild.id}/${message.channel.id}/${message.id}`,
+      "\n\n",
+      userNames && `Reactors: \`${userNames}\``,
+      staffNames && userNames && "\n",
+      staffNames && `Staff: \`${staffNames}\``
+    ]
+      .filter(Boolean)
+      .join("");
+
+    if (numberOfTotalReactions >= config.warningThreshold) {
+      logMessage = `<@${message.author.id}> has met the warning threshold in <#${message.channel.id}> for the message:`;
+    }
+
+    if (numberOfStaffReactions >= config.deletionThreshold) {
+      logMessage = `\`<@${message.author.id}> has met the deletion threshold in <#${message.channel.id}> for the message:`;
```

Comment:
Is this an unclosed ` at the very start of the message? Not sure if that's intentional.

Reply:
Looks like it, I originally had it as `@moderator` so it wouldn't ping during testing. Nice catch :eyes: